paper/src/main/java/net/draycia/carbon/paper/integration/fuuid/AllianceChannel.java

```diff
@@ -0,0 +1,127 @@
+import org.checkerframework.framework.qual.DefaultQualifier;
+import org.spongepowered.configurate.objectmapping.ConfigSerializable;
+
+import static net.draycia.carbon.api.channels.ChannelPermissionResult.channelPermissionResult;
+
+@DefaultQualifier(NonNull.class)
+@ConfigSerializable
+public class AllianceChannel extends AbstractFactionsChannel {
+
+    public static final String FILE_NAME = "factionsuuid-alliancechat.conf";
+
+    private transient @MonotonicNonNull @Inject CarbonMessages messages;
+    private transient @MonotonicNonNull @Inject UserManager<?> users;
+
+    public AllianceChannel() {
+        this.key = Key.key("carbon", "alliancechat");
+        this.commandAliases = List.of("ac");
+
+        this.messageSource = new ConfigChannelMessageSource();
+        this.messageSource.defaults = Map.of(
+            "default_format", "(alliance: %factionsuuid_faction_name%) <display_name>: <message>",
+            "console", "[alliance: %factionsuuid_faction_name%] <username>: <message>"
+        );
+    }
+
+    @Override
+    public ChannelPermissionResult speechPermitted(final CarbonPlayer player) {
+        return channelPermissionResult(
+            this.hasRelations(player, Relation.ALLY),
+            () -> this.messages.cannotUseFactionAllianceChannel(player)
+        );
+    }
+
+    @Override
+    public ChannelPermissionResult hearingPermitted(final CarbonPlayer player) {
+        return channelPermissionResult(
+            this.hasRelations(player, Relation.ALLY),
+            () -> this.messages.cannotUseFactionAllianceChannel(player)
+        );
+    }
+
+    @Override
+    public List<Audience> recipients(final CarbonPlayer sender) {
+        if (!this.hasRelations(sender, Relation.ALLY)) {
+            if (sender.online()) {
+                sender.sendMessage(this.messages.cannotUseFactionAllianceChannel(sender));
+            }
+
+            return Collections.emptyList();
+        }
+
+        final List<Audience> recipients = new ArrayList<>();
+        for (final Player player : this.alliedPlayersTo(sender)) {
+            final @Nullable CarbonPlayer carbon = this.users.user(player.getUniqueId()).getNow(null);
+            if (carbon != null) {
+                recipients.add(carbon);
+            }
+        }
+
+        recipients.add(this.server.console());
+
+        return recipients;
+    }
+
+    private List<Player> alliedPlayersTo(final CarbonPlayer player) {
+        final @Nullable Faction faction = this.faction(player);
+
+        if (faction == null) {
+            return List.of();
+        }
+
+        final List<Player> alliedPlayers = new ArrayList<>();
+
+        for (final FPlayer onlinePlayer : FPlayers.getInstance().getOnlinePlayers()) {
+            final Relation relation = faction.getRelationTo(onlinePlayer);
+
+            // Not sure if checking MEMBER is required, or if we only need to check ALLY
+            if (relation == Relation.MEMBER || relation == Relation.ALLY) {
```

Comment:
I think we should use `relation.isAtLeast(Relation.ALLY)` here

https://github.com/drtshock/Factions/blob/1.6.x/src/main/java/com/massivecraft/factions/perms/Relation.java#L101